microcosm_flask/conventions/encoding.py

```diff
@@ -170,3 +191,36 @@ def require_response_data(response_data):
     if not response_data:
         raise NotFound
     return response_data
+
+
+def csvify(response_data, response_schema):
+    """
+    Make Flask `response` object, with data returned as a generator for the CSV content
+    The CSV is built from JSON-like object (Python `dict` or list of `dicts`)
+
+    """
+    # TODO: determine column ordering with response schema
+    if "items" in response_data:
+        list_response_data = response_data["items"]
+    else:
+        list_response_data = [response_data]
+
+    response_fields = list(list_response_data[0].keys())
+
+    column_order = getattr(response_schema, "csv_column_order", None)
+    if column_order is None:
+        # We should still be able to return a CSV even if no column order has been specified
+        column_names = response_fields
+    else:
+        column_names = response_schema.csv_column_order
+        # The column order be only partially specified
+        column_names.extend([field_name for field_name in response_fields if field_name not in column_names])
+
+    output = StringIO()
+    csv_writer = writer(output, quoting=QUOTE_MINIMAL)
```

Comment:
I recommend testing this with a large data set. Memory usage could go up if we try to dump the entire result set into a string buffer. Might be ok, might not (since our containers usually have a memory cap)